### PR TITLE
Done Homework Test_Lecture

### PR DESCRIPTION
**Pull request by Chepak-Semen**

### CODE REVIEW

**Comment by Chepak-Semen** [comment]
[HW#3-TestChepak-Semen](https://trello.com/c/7eCoJle1/60-hw3-testchepak-semen)

File: homework/Homework_Test.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        self.assertEqual(test_simple_calc.subtract(9, 12), -3)
+
+    def test_multiply(self):
+        self.assertEqual(test_simple_calc.multiply(0, 2), 0)
+        self.assertEqual(test_simple_calc.multiply(4, -2), -8)
+        self.assertEqual(test_simple_calc.multiply(-4, -2), 8)
+
+    def test_divide(self):
+        self.assertEqual(test_simple_calc.divide(0, 2), 0)
+        self.assertRaises(ValueError, lambda: test_simple_calc.divide(2, 0))
```
Comment (oliaboo) [inline homework/Homework_Test.py:28]: separate test - test_divide_by_zero

File: homework/Homework_Test.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+"""Homework after tests-lecture:
+1. write unit tests for:
+ test_simple_calc.py + test_simple_employee.py
+"""
+import unittest
+import test_simple_calc
+import tests_simple_employee
+from unittest.mock import patch
+
+PERSON = tests_simple_employee.Employee('Andriy', 'Rocket', 10000)
```
Comment (oliaboo) [inline homework/Homework_Test.py:10]: move to either setUp or fixture (GLOBAL should not be widely used)

File: homework/Homework_Test.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    def test_fullname(self):
+        self.assertEqual(PERSON.fullname, 'Andriy Rocket')
+
+    def test_apply_raise(self):
+        PERSON.apply_raise()
+        self.assertEqual(PERSON.pay, 10500)
+
+    def test_monthly_schedule(self):
+        with patch('tests_simple_employee.requests.get') as mock:
+            mock.return_value.ok = False
```
Comment (oliaboo) [inline homework/Homework_Test.py:45]: add test for ok response

File: homework/Homework_test_2.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import unittest
+from tests_complex import Test, new_test, func
+from unittest import mock
+
+
+class TestForTestClass(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.m = mock.Mock()
+        self.TESTER = Test()
```
Comment (oliaboo) [inline homework/Homework_test_2.py:14]: `TESTER` - upper-case not needed

File: homework/Homework_test_2.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+"""
+import unittest
+from tests_complex import Test, new_test, func
+from unittest import mock
+
+
+class TestForTestClass(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.m = mock.Mock()
```
Comment (oliaboo) [inline homework/Homework_test_2.py:13]: `m` is a bad name, I should understand what for it stays from the name

File: homework/Homework_test_2.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        self.TESTER = Test()
+        self.m.exc_type = None
+        self.m.exc_val = None
+        self.m.exc_tb = None
+
+    def tearDown(self) -> None:
+        ok = self.defaultTestResult().wasSuccessful()
+        if ok:
+            print(f"{self._testMethodName} Was Successful : {self.defaultTestResult().wasSuccessful()}")
```
Comment (oliaboo) [inline homework/Homework_test_2.py:22]: did you thy making higher verbosity for unittest run?
not actually sure, this is needed.

File: homework/Homework_test_2.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+class TestFunc(unittest.TestCase):
+
+    def tearDown(self) -> None:
+        ok = self.defaultTestResult().wasSuccessful()
+        if ok:
+            print(f"{self._testMethodName} Was Successful : {self.defaultTestResult().wasSuccessful()}")
+
+    def test_func(self):
+        self.assertEqual(func(), 1)
```
Comment (oliaboo) [inline homework/Homework_test_2.py:55]: as an idea:
you may mock `new_test` function with:
```
mock.MagicMock(
        __enter__=<enter_mock>,
        __exit__=<exit_mock>
    )
```
and then check whether `__enter__` and `__exit__` have been called

**Review by oliaboo — CHANGES_REQUESTED** [review]

File: homework/.gitignore [hunk 8]
```diff
@@ -0,0 +1 @@
+*.idea
```
Comment (oliaboo) [inline homework/.gitignore:1]: https://github.com/github/gitignore/blob/master/Python.gitignore
here is the example for .gitignore file within Python project (at least, add `__pycache__`)

File: homework/Homework_Test.py (hunk 1 above)
Comment (dimigor) [reply homework/Homework_Test.py:26]: Зроби два окремі тести, коли вдале ділення, і коли з ексепшином

**Review by dimigor — CHANGES_REQUESTED** [review]

File: homework/Homework_Test.py (hunk 1 above)
Comment (Chepak-Semen) [reply homework/Homework_Test.py:26]: О'кей

**Review by dimigor — APPROVED** [review]

**Pull request merged**